app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,113 @@
+class TasksController < ApplicationController
+  
+  
+  def index
+    @greeting = "HELLO, THERE!"
+    @tasks = Task.all
+  end
+  
+  
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      flash[:error] = "Could not find task with id: " + task_id.to_s
+      redirect_to root_path
+      return
+    end
+  end
+  
+  
+  def new
+    @task = Task.new
+    @task.completion_date = nil
+  end
+  
+  
+  def create
+    @task = Task.new( task_params )
+    
+    @task.save
+    
+    if @task.save
```

Comment:
You end up double-saving here in this method... which shouldn't have any effect, but it's worth noting that you probably just meant to save it once!